### PR TITLE
Future will fail when the handler which is passed to `Future.future` throws an Exception.

### PR DESCRIPTION
**Pull request by cpdyj**
I means Future can auto fail when handler which is passed to `Future.future` throws an Exception.
This is my first PR. 
See also: #3056 
Thank you.

### CODE REVIEW

**Comment by cpdyj** [comment]
I modify the code, But it break the test case. The testcase need update. Should I update it? 
Thank you.

File: src/main/java/io/vertx/core/Future.java [hunk 1]
```diff
@@ -36,7 +36,11 @@
    */
   static <T> Future<T> future(Handler<Promise<T>> handler) {
     Promise<T> promise = Promise.promise();
-    handler.handle(promise);
+    try{
+      handler.handle(promise);
+    }catch (Exception e){
```
Comment (vietj) [inline src/main/java/io/vertx/core/Future.java]: missing whitespace

File: src/main/java/io/vertx/core/Future.java [hunk 2]
```diff
@@ -36,7 +36,11 @@
...
+      promise.fail(e);
```
Comment (vietj) [inline src/main/java/io/vertx/core/Future.java]: use tryFail instead

**Comment by vietj** [comment]
the test should indeed be modified so that it does not fail after and check that the returned promise is actually failed with the same exception that is thrown.

in addition we need a test that check that if the promise is resolved (like a completion) before the exception is thrown then the outcome of the promise resolution is not modified

**Comment by vietj** [comment]
you need to sign the Eclipse Contributor Agreeement and sign your commits as well

File: src/main/java/io/vertx/core/Future.java (hunk 1 above)
Comment (cpdyj) [reply src/main/java/io/vertx/core/Future.java]: I will reformat code.

**Comment by cpdyj** [comment]
Ok, I will modify code. Thank you.

**Comment by cpdyj** [comment]
@vietj Ok, I finished.

**Pull request merged**